src/test/java/com/proofpoint/discovery/TestIpHostnameAuthManager.java

```diff
@@ -0,0 +1,333 @@
+import org.testng.annotations.Test;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.ws.rs.ForbiddenException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.UUID;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+import static com.proofpoint.audit.AuditLoggerBinder.auditLoggerBinder;
+import static com.proofpoint.bootstrap.Bootstrap.bootstrapTest;
+import static com.proofpoint.configuration.ConfigBinder.bindConfig;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatCode;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class TestIpHostnameAuthManager
+{
+    private final ServiceDescriptor discoveryDescriptor = new ServiceDescriptor(
+            UUID.randomUUID(),
+            UUID.randomUUID().toString(),
+            "discovery",
+            "general",
+            "/location",
+            ServiceState.RUNNING,
+            ImmutableMap.of("http", "http://localhost:4111"));
+    private final DiscoverySelector selector = new DiscoverySelector().add(discoveryDescriptor);
+    private final TestingTimeSupplier timeSupplier = new TestingTimeSupplier();
+    private final DiscoveryConfig discoveryConfig = new DiscoveryConfig();
+    private InMemoryDynamicStore dynamicStore;
+    private SerialScheduledExecutorService executor;
+    private LifeCycleManager lifeCycleManager;
+    private AuthManager authManager;
+    private TestingAuditLog auditLog;
+
+    @BeforeMethod
+    public void beforeMethod()
+            throws Exception
+    {
+        dynamicStore = new InMemoryDynamicStore(discoveryConfig, timeSupplier);
+        executor = new SerialScheduledExecutorService();
+        Injector injector = bootstrapTest()
+                .withModules(
+                        new JsonModule(),
+                        new TestingAuditLogModule(),
+                        (binder -> {
+                            binder.bind(DynamicStore.class).toInstance(dynamicStore);
+                            binder.bind(ServiceSelector.class).toInstance(selector);
+                            bindConfig(binder).bind(StoreConfig.class);
+                            binder.bind(ScheduledExecutorService.class).annotatedWith(ForAuthManager.class).toInstance(executor);
+                            auditLoggerBinder(binder).bind(AuthAuditRecord.class);
+                            binder.bind(AuthManager.class).to(IpHostnameAuthManager.class).in(Scopes.SINGLETON);
+                        }))
+                .initialize();
+        lifeCycleManager = injector.getInstance(LifeCycleManager.class);
+        authManager = injector.getInstance(AuthManager.class);
+        auditLog = injector.getInstance(TestingAuditLog.class);
+    }
+
+    @AfterMethod
```

Comment:
```suggestion
    @AfterMethod(alwaysRun = true)
```
but then `TestDiscoveryServer` has the same problem.